redux/react/ConnectedComponentMacro.hx

```diff
@@ -0,0 +1,665 @@
+package redux.react;
+
+import haxe.EnumTools.EnumValueTools;
+import haxe.macro.ComplexTypeTools;
+import haxe.macro.Context;
+import haxe.macro.Expr;
+import haxe.macro.Type;
+import haxe.macro.TypeTools;
+
+@:access(react.ReactMacro)
+class ConnectedComponentMacro {
+	static function buildContainer():Array<Field>
+	{
+		var inClass = Context.getLocalClass().get();
+		var fields = Context.getBuildFields();
+
+		var componentType:Type = null;
+		var propsType:ComplexType = macro :Dynamic;
+
+		switch (inClass.superClass)
+		{
+			case {params: params, t: _.toString() => 'redux.react.ReactContainerFor'}:
+				componentType = params[0];
+				propsType = TypeTools.toComplexType(params[1]);
+
+			default:
+				Context.fatalError('You cannot inherit from a react container', inClass.pos);
+		}
+
+		checkNonStaticFields(fields);
+		checkReservedFields(fields);
+
+		var componentPropsType = getComponentPropsType(componentType, inClass);
+		var connectedFunctions = getConnectedFunctions(fields, componentPropsType, inClass);
+
+		var mapStateToProps = getMapStateToProps(fields, propsType, componentPropsType);
+		var mapDispatchToProps = getMapDispatchToProps(fields, propsType, componentPropsType);
+		var mergeProps = getMergeProps(fields, propsType, componentPropsType);
+		var options = getOptions(fields);
+
+		if (mapDispatchToProps == null)
+			addMapDispatchToProps(
+				fields,
+				connectedFunctions,
+				propsType,
+				componentPropsType,
+				inClass
+			);
+		else if (connectedFunctions.length > 0)
+			addMapDispatchToPropsWarnings(connectedFunctions);
+
+		if (mapStateToProps == null) addMapStateToProps(fields, inClass);
```

Comment:
addMapStateToProps, addMergeProps and addOptions only create `null` objects so that 
```
redux.react.ReactRedux.connect(
					mapStateToProps,
					mapDispatchToProps,
					mergeProps,
					options
				)(render);
```
does not get any `undefined` (you can see them at the very bottom of the file)